taskRequests/script.js

```diff
@@ -0,0 +1,188 @@
+const API_BASE_URL = window.API_BASE_URL;
+const taskRequestContainer = document.getElementById('task-request-container');
+const containerBody = document.querySelector('.container__body');
+const filterContainer = document.querySelector('.container__filters');
+
+const fetchedTaskRequests = [];
+
+const loader = document.querySelector('.container__body__loader');
+const startLoading = () => loader.classList.remove('hidden');
+const stopLoading = () => loader.classList.add('hidden');
+
+function createCustomElement(domObjectMap) {
+  const el = document.createElement(domObjectMap.tagName);
+  for (const [key, value] of Object.entries(domObjectMap)) {
+    if (key === 'tagName') {
+      continue;
+    }
+    if (key === 'eventListeners') {
+      value.forEach((obj) => {
+        el.addEventListener(obj.event, obj.func);
+      });
+    }
+    if (key === 'class') {
+      if (Array.isArray(value)) {
+        el.classList.add(...value);
+      } else {
+        el.classList.add(value);
+      }
+    } else if (key === 'child') {
+      el.append(...value);
+    } else {
+      el[key] = value;
+    }
+  }
+  return el;
+}
+
+async function getTaskRequests() {
+  startLoading();
+  try {
+    const res = await fetch(`${API_BASE_URL}/taskRequests`);
+
+    if (res.ok) {
+      const data = await res.json();
+      fetchedTaskRequests.push(...data.data);
+      return;
+    }
+
+    if (res.status === 401) {
+      showMessage('ERROR', ErrorMessages.UNAUTHENTICATED);
+      return;
+    }
+
+    if (res.status === 403) {
+      showMessage('ERROR', ErrorMessages.UNAUTHORIZED);
+      return;
+    }
+
+    if (res.status === 404) {
+      showMessage('ERROR', ErrorMessages.NOT_FOUND);
+      return;
+    }
+
+    showMessage('ERROR', ErrorMessages.SERVER_ERROR);
```

Comment:
Nit: for status like ERROR, SUCCESS, better to use a constant and reuse it everywhere rather than manually typing it everywhere

Reply:
Currently this utility is only used for this component, I don't think over engineering it with constants would be ideal as this might change once we decide common utilities across website-dashboard.